### ISSUE TITLE
[BUG] Missing stack trace of thrown error in log produced by nestjs built-in error handler

### ISSUE THREAD

**Issue by raidokaldma** [iamolegga/nestjs-pino#656]
[x] I've read [the docs of nestjs-pino](https://github.com/iamolegga/nestjs-pino/blob/master/README.md)

[x] I've read [the docs of pino](https://getpino.io/#/)

[x] I couldn't find the same [open issue of nestjs-pino](https://github.com/iamolegga/nestjs-pino/issues)

There is an open issue https://github.com/iamolegga/nestjs-pino/issues/535, but that issue focuses on `express`, but I am not convinced my issue is related to `express`.

**What is the current behavior?**
`nestjs-pino@2.3.0` is missing stack traces

**What is the expected behavior?**
Stack traces should be included in error logs, just like previously in `nestjs-pino@1.4.0`

**Please provide minimal example repo. Without it this issue will be closed**
I created two NestJS applications, older one includes stack traces in error logs, newer version does not.

See examples here: https://github.com/raidokaldma/nestjs-pino-issue

**Comment by iamolegga** [comment]
Finally found the bug. Working on fixing it.

In short, there is a difference between nestjs built-in `LoggerService` interface in [v7](https://docs.nestjs.com/v7/techniques/logger) and [v8](https://docs.nestjs.com/techniques/logger). v7's `error` method has `stack` argument, but in v8 it's dropped for consistency between all the other methods. But [in many places of the nestjs sources](https://github.com/search?l=TypeScript&q=org%3Anestjs+logger+error+stack&type=Code) it's still called like in v7, so this becomes a headache for lib owners to detect if `optionalParams` contains stack to properly set such field when logging error 😅

This issue differs from #535 because here the reason is a difference between logger versions and the logger call in exception handlers, and #535 is about `request completed/errored` `pino-http`'s log

**Comment by iamolegga** [comment]
shipped in v2.3.1

**Comment by raidokaldma** [comment]
I verified the fix works as expected. Thank you! 🙇

**Comment by raidokaldma** [comment]
> In short, there is a difference between nestjs built-in `LoggerService` interface in [v7](https://docs.nestjs.com/v7/techniques/logger) and [v8](https://docs.nestjs.com/techniques/logger). v7's `error` method has `stack` argument, but in v8 it's dropped for consistency between all the other methods. But [in many places of the nestjs sources](https://github.com/search?l=TypeScript&q=org%3Anestjs+logger+error+stack&type=Code) it's still called like in v7, so this becomes a headache for lib owners to detect if `optionalParams` contains stack to properly set such field when logging error sweat_smile

Just as a side note, It seems to me they did not drop the `stack` argument from built in logger's definition.
```
error(message: any, stack?: string, context?: string): void;
```
See: https://github.com/nestjs/nest/blob/v8.0.0/packages/common/services/logger.service.ts#L108

**Comment by iamolegga** [comment]
But that's internal implementation. You should look at public interface for logger [here](https://github.com/nestjs/nest/blob/v8.0.0/packages/common/services/logger.service.ts#L18) and in the [docs](https://docs.nestjs.com/techniques/logger#custom-implementation)

**Comment by raidokaldma** [comment]
In short, `Logger` is exported class, so it is not internal. However, if you wanted to extend their LoggerService, then you would be correct, `error` method is missing `stack` argument.

In my case, I am actually using `Logger` in my application as in example here: https://docs.nestjs.com/techniques/logger#using-the-logger-for-application-logging

For example:
```
import { Logger } from '@nestjs/common';
...
class MyService {
  private readonly logger = new Logger(MyService.name);
  ...
}
```

And that's actually how I stumbled upon the issue.

Long story short, your fix is golden, I appreciate your quick response and effort. Thank you.
